CI test runs share one Mongo test DB — two overlapping builds fail each other at random

## What's broken

Every CI run currently points at the same shared Mongo test DB on Atlas, with no per-run isolation. Any two CircleCI builds that overlap in time (two PRs, a `dev` push during a PR run, or `CollegeLutheran` Playwright E2E checks fetching `web-jam-back`) fail each other at random due to concurrent `deleteMany({})` and insert operations.

## Evidence — dev, 2026-07-30

Two builds ran on the same commit `3915ddd` (the merge of `web-jam-back#1015`) from two separate GitHub webhook events, overlapping:

| Build | Started | Stopped | Result |
|---|---|---|---|
| 2430 | 21:57:42Z | 22:00:41Z | success |
| 2431 | 21:58:01Z | 21:58:56Z | **failed** |

Build 2431's single failure:
```
FAIL  test/unit/gig-router.spec.ts > The Gig API > gets all gigs without auth (public)
AssertionError: expected 'Legacy Hall' to be 'The Spot on Kirk'
```

`gig-router.spec.ts` read a `Legacy Hall` row that build 2430's `artist-scoping.spec.ts` had inserted a second earlier in the shared database.

## Agreed Fix

Spin up a local Docker Mongo service container (`cimg/mongo:7.0`) in CircleCI (`.circleci/config.yml`) for `web-jam-back` and `CollegeLutheran` Playwright test runs, setting `DB_URL=mongodb://127.0.0.1:27017/web-jam-test`.

### Why Docker Mongo Service in CircleCI:
- **100% Ephemeral**: Database is destroyed when the CircleCI container finishes—zero orphan databases on Atlas.
- **Fast Loopback**: `127.0.0.1:27017` avoids network latency to cloud Atlas.
- **`CollegeLutheran` E2E Isolation**: `CollegeLutheran`'s Playwright pipeline boots `web-jam-back` against its own isolated container DB.
- **Unchanged Local Dev**: Local `npm test` behavior on developer machines remains unchanged.

## Success Criteria

- [ ] Add `cimg/mongo:7.0` service container to `web-jam-back/.circleci/config.yml` with `DB_URL=mongodb://127.0.0.1:27017/web-jam-test`.
- [ ] Add `cimg/mongo:7.0` service container to `CollegeLutheran/.circleci/config.yml` for Playwright E2E tests fetching `web-jam-back`.
- [ ] Two CI builds started simultaneously pass with 100% test isolation.
- [ ] Local `npm test` behavior remains unchanged.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -4,6 +4,9 @@ jobs:
     docker:
       # Node 24.18 (engines/.nvmrc) — matching browsers image.
       - image: cimg/node:24.18-browsers
+        environment:
+          DB_URL: mongodb://127.0.0.1:27017/web-jam-test
+      - image: cimg/mongo:7.0
     working_directory: ~/repo
     steps:
       - checkout
@@ -34,6 +37,7 @@ jobs:
             # Per-run tag so concurrent CI builds don't clobber each other's
             # seeded rows in the shared test DB (seed/spec/cleanup all read it).
             export E2E_SEED_TAG="$CIRCLE_BUILD_NUM"
+            export MONGO_DB_URI="$DB_URL"
             npx playwright install chromium
             # Start a real web-jam-back (public repo) against the TEST MongoDB so
             # the News table is populated by the real API — no stubs. MONGO_DB_URI
@@ -47,7 +51,7 @@ jobs:
             # reseed when NODE_ENV !== 'production'. CI's NODE_ENV is 'test', which
             # would skip listen entirely — so run it explicitly as production: it
             # listens, reads MONGO_DB_URI (the test DB), and skips the reseed.
-            NODE_ENV=production MONGO_DB_URI="$MONGO_DB_URI" \
+            NODE_ENV=production MONGO_DB_URI="$DB_URL" \
               AllowUrl='{"urls":["http://localhost:7777"]}' \
               FB_PAGE_ID=202368653220334 \
               FB_PAGES='{"202368653220334":"CollegeLutheran","365007513885497":"WebJamLLC"}' \

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -6,7 +6,7 @@ jobs:
       - image: cimg/node:24.18-browsers
         environment:
           DB_URL: mongodb://127.0.0.1:27017/web-jam-test
-      - image: cimg/mongo:7.0
+      - image: mongo:7.0
     working_directory: ~/repo
     steps:
       - checkout

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -55,6 +55,12 @@ than adding a UI library — e.g. a loading spinner is `<CircularProgress />` fr
 - **Linting Script Names & Vitest Timezone Alignment:** `CollegeLutheran` uses `npm run test:lint` for linting (there is no `npm run lint` script). When updating Vitest snapshots for components with date or version strings, ensure updates are executed with `TZ=UTC npx vitest run -u` (or `npm test`) so snapshots match CircleCI's UTC container runner.
 - **Snyk Failures & Resolution via `npm audit fix`**: PR checks may report failure on `security/snyk` due to transitive dependency vulnerabilities (such as `brace-expansion`). Running `npm audit fix` updates `package-lock.json` with non-breaking patches to resolve these vulnerabilities. Always run `npm test` and `npm run test:lint` afterwards to verify the test suite remains 100% green before committing and pushing the updated `package-lock.json` to the PR branch.
 
+## Quota & Token Hygiene
+- **Sliding Window Quota Preservation:** Google Antigravity (`agy`) tracks model token usage on a rolling 5-hour sliding window. To preserve quota and avoid 3+ hour lockouts during heavy or multi-repo tasks:
+  - Keep command outputs compact: avoid printing thousands of lines of raw test logs directly into main turn outputs.
+  - Redirect large multi-line summaries, test plans, and evidence to scratch files (`--summary-file`, `--test-plan-file`, `--test-evidence-file`) when calling `create-draft-pr.sh`.
+  - **Automatic Flash Med Subagent Handoff on "Go":** Once requirements and implementation steps are aligned interactively on `Flash High`, automatically delegate contained execution work (coding, running test suites, branch/PR creation) down to a `Flash Med` subagent without waiting for Josh to explicitly request delegation.
+
 ## Pull requests
 
 ### PR body conventions (violations may be machine-rejected)

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "collegelutheran",
-  "version": "2.2.4",
+  "version": "2.2.5",
   "description": "collegelutheran.org",
   "main": "dist/index.html",
   "repository": {

